### PR TITLE
docs: surface new 1.12.0 features and enhancements 

### PR DESCRIPTION
**Pull request by heitorlessa**
**Issue #, if available:**

## Description of changes:

Surface new functionalities, enhancements, and refresh docs to better surface certain features previously hidden:

* **Parameters**: surface how to change cache TTL
* **Parameters**: add new force_fetch param
* **Parameters**: refresh to use new navigation
* **Idempotency**: add note on reusing a single table for idempotency state
* **Logger**: surface usage of new correlation_id and paths
* **Dataclasses**: remove additional note banners to ease reading
* **Dataclasses**: update AppSync Resolver to emphasize correlation, typos

**Checklist**

 

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [ ] Update tests
* [x] Update docs
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/376ec0a2ac0d2a40e0af5717bef42ff84ca0d1b9/.github/semantic.yml#L2)

## Breaking change checklist

**RFC issue #**:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by michaelbrewer** [comment]
@heitorlessa  

I do have some code example use cases for `force_fetch` in the PR description:
- https://github.com/awslabs/aws-lambda-powertools-python/pull/341

And a code example of the AppSync decorator:
- https://github.com/awslabs/aws-lambda-powertools-python/pull/323

File: docs/utilities/parameters.md [hunk 1]
```diff
@@ -88,9 +88,34 @@ The following will retrieve the latest version and store it in the cache.
 
 ## Advanced
 
+### Adjusting cache TTL
+
+By default, we cache parameters retrieved in-memory for 5 seconds.
+
+You can adjust how long we should keep values in cache by using the param `max_age`, when using  `get()` or `get_multiple` methods across all providers.
```
Comment (michaelbrewer) [inline docs/utilities/parameters.md:95]: @heitorlessa  maybe a typo?

```suggestion
You can adjust how long we should keep values in cache by using the param `max_age`, when using  `get()` or `get_multiple()` methods across all providers.
```

File: docs/utilities/parameters.md [hunk 2]
```diff
@@ -92,7 +92,7 @@ The following will retrieve the latest version and store it in the cache.
 
 By default, we cache parameters retrieved in-memory for 5 seconds.
 
-You can adjust how long we should keep values in cache by using the param `max_age`, when using  `get()` or `get_multiple` methods across all providers.
+You can adjust how long we should keep values in cache by using the param `max_age`, when using  `get()` or `get_multiple()` methods across all providers.
```
Comment (michaelbrewer) [inline docs/utilities/parameters.md:95]: @heitorlessa  haha. you got it

**Review by michaelbrewer — APPROVED** [review]
Looks cool. 

We might want to remove the `New in 1.12.0` in future releases as we might start to clutter the docs. It could be some kind of expiring things :) 

:shipit:

**Comment by heitorlessa** [comment]
> @heitorlessa
> 
> I do have some code example use cases for `force_fetch` in the PR description:
> 
>     * #341
> 
> 
> And a code example of the AppSync decorator:
> 
>     * #323

Thanks Michael - I found the first example being better suited for a new Guide section, as we'd want to keep it brief. 

The App Resolver decorator, there's a high chance (90%) this will be moved once the RFC #325 and #322 get more details, so let's keep this undocumented for now.

**Comment by heitorlessa** [comment]
> Looks cool.
> 
> We might want to remove the `New in 1.12.0` in future releases as we might start to clutter the docs. It could be some kind of expiring things :)
> 
> :shipit:

Actually there's going to be an even better news that this will no longer be necessary - Our documentation system now supports versioning, meaning we could have that for either per release tag OR per branch (less convoluted)

https://squidfunk.github.io/mkdocs-material/changelog/#700-_-february-22-2021

**Comment by michaelbrewer** [comment]
> > @heitorlessa
> > I do have some code example use cases for `force_fetch` in the PR description:
> > ```
> > * #341
> > ```
> > 
> > 
> > And a code example of the AppSync decorator:
> > ```
> > * #323
> > ```
> 
> Thanks Michael - I found the first example being better suited for a new Guide section, as we'd want to keep it brief.
> 
> The App Resolver decorator, there's a high chance (90%) this will be moved once the RFC #325 and #322 get more details, so let's keep this undocumented for now.

Sounds good. Maybe we have a cookbook guide somewhere that has more detailed use cases (or it could just be blog posts we refer to)

**Pull request merged**